domain/modeldefaults/state/state.go

```diff
@@ -81,93 +87,94 @@ WHERE m.uuid = ?
+	defaults := map[string]map[string]string{}
 
-	db, err := s.DB()
+	db, err := st.DB()
 	if err != nil {
-		return rval, errors.Trace(err)
+		return defaults, interrors.Errorf("getting database instance for cloud region defaults: %w", err)
 	}
 
-	cloudDefaultsStmt := `
-SELECT cloud_region_defaults.key,
-       cloud_region_defaults.value
+	stmt, err := st.Prepare(`
+SELECT (cloud_region.name, cloud_region_defaults.key, cloud_region_defaults.value) AS (&cloudRegionDefaultValue.*)
 FROM cloud_region_defaults
-INNER JOIN cloud_region
-ON cloud_region.uuid = cloud_region_defaults.region_uuid
-INNER JOIN model m
-ON m.cloud_region_uuid = cloud_region.uuid
-WHERE m.uuid = ?
-`
+JOIN cloud_region ON cloud_region.uuid = cloud_region_defaults.region_uuid
+WHERE cloud_region.cloud_uuid = $dbCloud.uuid
+`, cloudRegionDefaultValue{}, dbCloud{})
+	if err != nil {
+		return defaults, errors.Trace(err)
+	}
 
-	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		rows, err := tx.QueryContext(ctx, cloudDefaultsStmt, uuid)
-		if err != nil {
-			return interrors.Errorf("fetching cloud region defaults for model %q: %w", uuid, err)
+	return defaults, db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+
+		var regionDefaultValues []cloudRegionDefaultValue
+
+		if err := tx.Query(ctx, stmt, dbCloud{UUID: cloudUUID.String()}).GetAll(&regionDefaultValues); err != nil {
+			if errors.Is(err, sqlair.ErrNoRows) {
+				return nil
+			}
+			return interrors.Errorf("fetching cloud %q region defaults: %w", cloudUUID, err)
 		}
-		defer rows.Close()
 
-		var (
-			key, val string
-		)
-		for rows.Next() {
-			if err := rows.Scan(&key, &val); err != nil {
-				return interrors.Errorf("reading cloud region defaults for model %q: %w", uuid, err)
+		for _, regionDefaultValue := range regionDefaultValues {
+			store, has := defaults[regionDefaultValue.RegionName]
+			if !has {
+				store = map[string]string{}
+				defaults[regionDefaultValue.RegionName] = store
 			}
-			rval[key] = val
+			store[regionDefaultValue.Key] = regionDefaultValue.Value
 		}
-		return rows.Err()
+		return nil
 	})
-
-	if err != nil {
-		return nil, err
-	}
-	return rval, nil
 }
 
-// ModelCloudType returns the cloud type for model identified by the given model
-// uuid. If no model exists for the provided model uuid then an error satisfying
-// [modelerrors.NotFound] is returned.
-func (s *State) ModelCloudType(
+// CloudType returns the cloud type uof the cloud.
+// If no cloud exists for the given uuid then an error
+// satisfying [clouderrors.NotFound] is returned.
+func (s *State) CloudType(
 	ctx context.Context,
-	uuid coremodel.UUID,
+	uuid cloud.UUID,
 ) (string, error) {
 	db, err := s.DB()
 	if err != nil {
 		return "'", errors.Trace(err)
 	}
 
-	modelUUIDVal := modelUUIDValue{UUID: uuid.String()}
+	cld := dbCloud{UUID: uuid.String()}
 	result := modelCloudType{}
 
 	stmt, err := s.Prepare(`
 SELECT (ct.type) AS (&modelCloudType.cloud_type)
```

Comment:
The brackets are not needed round here, you can just write `SELECT ct.type AS &modelCloudType.cloud_type`

Reply:
Yes indeed, thanks. This was copied across. Fixed.